submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+            case 'alphabet':
+                uncheckRadio('age');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAlphabet
+                ]);
+
+                break;
+            case 'gender':
+                const pipes = [
+                    filterByName,
+                    filterByGender,
+                ];
+                if (getOpt('alphabet')) {
+                    pipes.push(sortByAlphabet);
+                }
+                if (getOpt('age')) {
+                    pipes.push(sortByAge);
+                }
+                renderUsers(pipes);
+
+                break;
+            case 'age':
+                uncheckRadio('alphabet');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAge
+                ]);
+                break;
+            default:
+                // renderUsers();
+        }
+    });
+});
+
+const filterByName = (user) => {
+    const searchValue = nameInput.value.trim().toLowerCase();
+    return user.name.first.toLowerCase().includes(searchValue);
+}
+
+const filterByGender = (user) => {
+    const genderRadioOpt = getOpt('gender');
+    if (!genderRadioOpt) { // Pass all if none gender checked
+        return true;
+    }
+
+    if (!['male', 'female'].includes(genderRadioOpt.id)) { // Pass all if 'all' gender checked
+        return true;
+    }
+
+    return genderRadioOpt.id === user.gender;
+}
+
+const sortByAlphabet = (a, b) => {
+    const alphabetRadioOpt = getOpt('alphabet');
+
+    if (alphabetRadioOpt) {
+        let isReversed = alphabetRadioOpt.id === 'alphabetZA';
+        const nameA = a.name.first.toUpperCase();
+        const nameB = b.name.first.toUpperCase();
+        let result;
+
+        // Compare
+        if (nameA < nameB) {
+            result = -1;
+        } else if (nameA > nameB) {
+            result = 1;
+        } else {
+            result = 0;
+        }
+
+        // Reverse if needed
+        if (isReversed) {
+            result *= -1;
+        }
+        return result;
+    }
+
+    return 0;
+}
+
+const sortByAge = (userA, userB) => {
+    const ageRadioOpt = getOpt('age');
+
+    if (ageRadioOpt) {
+        let isReversed = ageRadioOpt.id === 'ageHighest';
+        return isReversed
+            ? userB.dob.age - userA.dob.age
+            : userA.dob.age - userB.dob.age;
+    }
+
+    return 0;
+}
+
+//reset
+let filterReset = document.querySelector('.filter__block button');
+
+filterReset.addEventListener('click', (e) => {
+    const value = e.preventDefault();
+    const genderAll = document.getElementById('genderAll');
+
+    closeUser();
+    genderAll.checked = true;
+    uncheckRadio('age');
+    uncheckRadio('alphabet');
+    filter.value = '';
+
+    renderUsers([]);
+});
+
+//close user
+const closeUser = () => {
+    let singleUser = document.querySelector('.user--single');
```

Comment:
If you've chosen a coding style - stick to it. Your CSS styles have `__` in them, why not `user__single`?

Reply:
changed to .user.user_single as modificatory